### ISSUE TITLE
Java?

### ISSUE THREAD

**Issue by EchedeyLR** [deltachat-bot/echo#9]
Whats about implement the same bot in Java?

**Comment by Simon-Laux** [comment]
good idea, but unfortunately the java bindings are not standalone, they are part of the android client.

**Comment by r10s** [comment]
the java bindings are still separated, they are in the folder https://github.com/deltachat/deltachat-android/tree/master/src/com/b44t/messenger 

@EchedeyLR so, if you want to push this forward, you can copy this folder (together with the jni part) and use it. or just use the whole deltachat-android as a subproject. of course, i expect some things to be adapted. also a "high-level" api missing.

if there are actual bots in java, at some point, we can think over to separate this, however, this comes at costs, so the advantages and disadvantages should be weigh carefully.

**Comment by EchedeyLR** [comment]
Thank you. I will check. As I never tried to mount a bot before I expected an example in the language I can use and try to extend from it.

**Comment by Simon-Laux** [comment]
The following imports sound not so standalone to me:
```
DcEventCenter.java:
import org.thoughtcrime.securesms.util.Util;

DcChatlist.java:
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

DcContact.java:
import android.graphics.Color;
import android.util.Log;

DcContext.java:
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

DcMsg.java:
import android.util.Log;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import org.thoughtcrime.securesms.connect.ApplicationDcContext;
```

**Comment by r10s** [comment]
@Simon-Laux thanks for the list, i made a pr removing these dependencies, https://github.com/deltachat/deltachat-android/pull/1825
